decidim-assemblies/app/models/decidim/assembly.rb

```diff
@@ -28,12 +28,21 @@ class Assembly < ApplicationRecord
 
     has_many :features, as: :participatory_space, dependent: :destroy
 
+    has_many :assembly_private_users, class_name: "Decidim::AssemblyPrivateUser", foreign_key: "decidim_assembly_id", dependent: :destroy
+    has_many :users, through: :assembly_private_users, class_name: "Decidim::User", foreign_key: "decidim_user_id"
+
     validates :slug, uniqueness: { scope: :organization }
     validates :slug, presence: true, format: { with: Decidim::Assembly.slug_format }
 
     mount_uploader :hero_image, Decidim::HeroImageUploader
     mount_uploader :banner_image, Decidim::BannerImageUploader
 
+    scope :visible_assemblies_for, lambda { |user|
```

Comment:
You can rename this to `visible_for`, since it's clear it will return assemblies.